subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -941,14 +1118,52 @@ static void pe_sender_response_run(void *obj)
 		 * Handle Sender Response Timeouts
 		 */
 		switch (current_state) {
+#if CONFIG_USBC_CSM_SINK_ONLY
 		/* Sink states */
 		case PE_SNK_SELECT_CAPABILITY:
 			pe_set_state(dev, PE_SNK_HARD_RESET);
 			break;
 		case PE_SNK_GET_SOURCE_CAP:
 			pe_set_state(dev, PE_SNK_READY);
 			break;
-
+#else
+		/* Source states */
+		case PE_SRC_DISCOVERY:
```

Comment:
Instead of putting these state-specific things in this super state, which will need to get bigger and more complicated over time, what if the super state set a flag when the timer expired, and each child state checked for it?

Reply:
This would add delay because the state machine will need to cycle again in order for the flag to be detected.